onnxruntime/test/providers/qnn/gather_nd_op_test.cc

```diff
@@ -0,0 +1,251 @@
+                              {1, 0},                                           // indices
+                              true,                                             // indices_are_static
+                              1,                                                // batch_dims
+                              ExpectedEPNodeAssignment::All, "GatherND_f32_StaticIndices_BatchDim1");
+}
+
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(__linux__)
+//
+// HTP tests:
+//
+
+// Test GatherND op on CPU backend:
+// positive, dynamic indices.
+// QNN EP's HTP backend does not support int64 data types.
+// Thefore, HTP does not support Dynamic int64_t indices at all.
+TEST_F(QnnHTPBackendTests, GatherND_f32_DynamicIndices_BatchDim0) {
+  RunHTPGatherNDOpTest<float>({2, 2, 2},                                        // data_shape
+                              {0.f, 0.1f, 0.2f, 0.3f, 0.4f, 0.5f, 0.6f, 0.7f},  // data
+                              {2, 1, 2},                                        // indices_shape
+                              {0, 1, 1, 0},                                     // indices
+                              false,                                            // indices_are_static
+                              0,                                                // batch_dims
+                              ExpectedEPNodeAssignment::Some,                   // QDQ GatherND not assigned to QNN EP.
+                              "GatherND_f32_DynamicIndices_BatchDim0",
+                              5);  // 2 QNN nodes (Quantize and Dequantize), 3 CPU EP nodes (QDQ GatherND).
+}
+
+// Test GatherND op on HTP backend:
+// positive, static indices.
+// HTP does not support int64, but QNN EP converts static int64 indices into int32.
+TEST_F(QnnHTPBackendTests, GatherND_u8_StaticIndices_BatchDim0) {
+  RunHTPGatherNDOpTest<float, uint8_t>({2, 2, 2},                                        // data_shape
+                                       {0.f, 0.1f, 0.2f, 0.3f, 0.4f, 0.5f, 0.6f, 0.7f},  // data
+                                       {2, 1, 2},                                        // indices_shape
+                                       {0, 1, 1, 0},                                     // indices
+                                       true,                                             // indices_are_static
+                                       0,                                                // batch_dims
+                                       ExpectedEPNodeAssignment::All, "GatherND_u8_StaticIndices_BatchDim0");
+}
+
+// Test GatherND op on HTP backend:
+// - positive, static indices.
+// - batch_dims = 1
+// QNN EP should support by converting static weights to int32_t.
+//
+// TODO: Enable when QNN fixes GatherNd with batch_dims != 0
+// Expected value: [[0.2, 0.3],[0.4, 0.5]], Actual (incorrect) output: [[0.2, 0.3], [0.0, 0.1]]
+TEST_F(QnnHTPBackendTests, DISABLED_GatherND_f32_StaticIndices_BatchDim1) {
```

Comment:
This unit test demonstrates potential QNN bug: Incorrect results when the batch_dims parameter is non-zero.